dvc/repo/experiments/show.py

```diff
@@ -190,13 +190,13 @@ def _sort_column(
             matches.add((path, sort_name, "metrics"))
         if path in param_names and sort_name in param_names[path]:
             matches.add((path, sort_name, "params"))
-    else:
+    if not matches:
         for path in metric_names:
-            if sort_name in metric_names[path]:
-                matches.add((path, sort_name, "metrics"))
+            if sort_by in metric_names[path]:
+                matches.add((path, sort_by, "metrics"))
         for path in param_names:
-            if sort_name in param_names[path]:
-                matches.add((path, sort_name, "params"))
+            if sort_by in param_names[path]:
```

Comment:
Do we want to support it in `--set-param` or are we implying that this will only be supported for exp show / metrics? 

(in the current fix we are also supporting params in the exp show)

Reply:
I think it's worth discussing, but I figured even this limited fix is better than the current situation.